### PR TITLE
scala error log organize

### PR DESCRIPTION
**Pull request by dding3**
In our current implementation, there are lots of `assert` , `require` in the code and sometimes doesn't provide much useful information. This pr adds scala error handling apis, and replace all existing `require` and `assert` with new apis, providing useful information.

below are some sample error message:
1. 
22-03-29 10:09:47 [ScalaTest-run-running-TrainingSpec] ERROR Log4Error$:40 - 

****************************Usage Error************************
Dim ordering must be either tf or th, but got test

****************************How to fix*************************
Please set dimOrdering=tf or dimOrdering=tf

****************************Call Stack*************************

Dim ordering must be either tf or th, but got test
java.lang.IllegalArgumentException: Dim ordering must be either tf or th, but got test
	at com.intel.analytics.bigdl.dllib.utils.Log4Error$.invalidInputError(Log4Error.scala:27)
	at com.intel.analytics.bigdl.dllib.keras.layers.utils.KerasUtils$.toBigDLFormat(KerasUtils.scala:164)
	at com.intel.analytics.bigdl.dllib.keras.layers.Convolution2D$.apply$mFc$sp(Convolution2D.scala:131)
	at com.intel.analytics.bigdl.dllib.keras.models.TrainingSpec$$anonfun$2.apply$mcV$sp(TrainingSpec.scala:101)
	at com.intel.analytics.bigdl.dllib.keras.models.TrainingSpec$$anonfun$2.apply(TrainingSpec.scala:99)
	at com.intel.analytics.bigdl.dllib.keras.models.TrainingSpec$$anonfun$2.apply(TrainingSpec.scala:99)

2.
****************************Usage Error************************
Total size after reshape must be unchanged. But in Reshapecc3d9897: input size is: 144, while reshape size is: 169
22-03-29 10:18:31 [ScalaTest-run-running-TrainingSpec] ERROR Log4Error$:47 - 

****************************Call Stack*************************
2022-03-29 10:18:31 INFO  SparkContext:54 - SparkContext already stopped.

Total size after reshape must be unchanged. But in Reshapecc3d9897: input size is: 144, while reshape size is: 169
com.intel.analytics.bigdl.dllib.utils.UnKnownException: Total size after reshape must be unchanged. But in Reshapecc3d9897: input size is: 144, while reshape size is: 169
	at com.intel.analytics.bigdl.dllib.utils.Log4Error$.unKnowExceptionError(Log4Error.scala:54)
	at com.intel.analytics.bigdl.dllib.keras.layers.Reshape.computeOutputShape(Reshape.scala:85)
	at com.intel.analytics.bigdl.dllib.nn.abstractnn.InferShape$class.build(InferShape.scala:76)
	at com.intel.analytics.bigdl.dllib.nn.internal.KerasLayer.build(KerasLayer.scala:317)
	at com.intel.analytics.bigdl.dllib.nn.internal.KerasLayer.inputs(KerasLayer.scala:334)

### CODE REVIEW

**Comment by jason-dai** [comment]
LGTM; can we add a style check in Github Action to enforce that no `assert` or `require` is allowed in the source code?

**Comment by dding3** [comment]
most jekins pass. http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-PR-Dllib-Validation/381/. The failed one(http://10.112.231.51:18888/job/BigDL-PRVN-Scala-StyleCheck/762//consoleFull) is not related to this pr.

**Comment by dding3** [comment]
http://10.112.231.51:18888/job/BigDL-PRVN-Scala-StyleCheck/765/console  pass
